sdk/projects_test.go

```diff
@@ -102,34 +100,95 @@ func TestProjectsGet(t *testing.T) {
 	})
 }
 
-func testCreateProjectClient(t *testing.T, url string) IProjects {
-	t.Helper()
+func TestGetProjectByID(t *testing.T) {
+	projectsListResponseBody := readTestData(t, "projects.json")
+	projectRequestAssertions := func(t *testing.T, req *http.Request) {
+		t.Helper()
 
-	client, err := jsonclient.New(jsonclient.Options{
-		BaseURL: url,
-		Headers: jsonclient.Headers{
-			"cookie": "sid=my-random-sid",
-		},
+		require.True(t, strings.HasSuffix(req.URL.Path, "/projects/"))
+		require.Equal(t, http.MethodGet, req.Method)
+		cookieSid, err := req.Cookie("sid")
+		require.NoError(t, err)
+		require.Equal(t, &http.Cookie{Name: "sid", Value: "my-random-sid"}, cookieSid)
+	}
+
+	t.Run("Error creating request for projectId fetch", func(t *testing.T) {
+		client := testCreateClient(t, "this-url-does-not-exist")
+		project, err := getProjectByID(client, "project1")
+		require.Nil(t, project)
+		require.EqualError(t, err, fmt.Sprintf("BaseURL must have a trailing slash, but \"this-url-does-not-exist\" does not"))
 	})
-	require.NoError(t, err, "error creating client")
 
-	return ProjectsClient{
-		JSONClient: client,
-	}
-}
+	t.Run("Network error occurs during projectId fetch", func(t *testing.T) {
```

Comment:
Why network error? This is an http error, not a netword error

Reply:
what do you mean? I can specify it's an HTTP error but HTTP errors are network errors 🤔

Reply:
Anyhow since it's a test case name I've made it much more explicit :)